File: Factory/Java/LogisticsNoFactory/src/LogisticsNoFactory/Truck.java
package LogisticsNoFactory;

import LogisticsNoFactory.Coordinate;

public class Truck {

	private String _whatToDeliver;
	private Coordinate[] _transportPoints;
	
	public Truck(String what, Coordinate[] transportPoints) {
		_whatToDeliver = what;
		_transportPoints = transportPoints;
	}
	
	public String transport(){
		StringBuilder builder = new StringBuilder("Transporting by truck through points\n");
		for(Coordinate c : _transportPoints)
			builder.append(c.toString());
		
		return builder.toString();
		
	}
	
	public String deliver()
	{
		return String.format("Delivering %s by Truck",_whatToDeliver);
	}
}

File: Factory/C#/Logistics/NoFactory/Truck.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NoFactory
{
    public class Truck
    {
        private readonly string _whatToDeliver;
        private readonly Coordinate[] _transportPoints;

        public Truck(string what, Coordinate[] transportPoints)
        {
            _whatToDeliver = what;
            _transportPoints = transportPoints;
        }

        public string Transport()
        {
            var builder = new StringBuilder("Transporting by truck through points\n");

            foreach (var transportPoint in _transportPoints)
            {
                builder.Append(transportPoint);
            }

            return builder.ToString();

        }

        public string Deliver()
        {
            return $"Delivering {_whatToDeliver}s by truck";
        }
    }
}
